ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/customercenter/InternalCustomerCenter.kt

```diff
@@ -0,0 +1,174 @@
+package com.revenuecat.purchases.ui.revenuecatui.customercenter
+
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.revenuecat.purchases.CacheFetchPolicy
+import com.revenuecat.purchases.CustomerInfo
+import com.revenuecat.purchases.ExperimentalPreviewRevenueCatPurchasesAPI
+import com.revenuecat.purchases.Offerings
+import com.revenuecat.purchases.PurchaseParams
+import com.revenuecat.purchases.PurchaseResult
+import com.revenuecat.purchases.PurchasesAreCompletedBy
+import com.revenuecat.purchases.PurchasesError
+import com.revenuecat.purchases.PurchasesErrorCode
+import com.revenuecat.purchases.customercenter.CustomerCenterConfigData
+import com.revenuecat.purchases.paywalls.events.PaywallEvent
+import com.revenuecat.purchases.ui.revenuecatui.customercenter.data.CustomerCenterState
+import com.revenuecat.purchases.ui.revenuecatui.customercenter.data.CustomerCenterViewModel
+import com.revenuecat.purchases.ui.revenuecatui.customercenter.data.CustomerCenterViewModelFactory
+import com.revenuecat.purchases.ui.revenuecatui.customercenter.data.CustomerCenterViewModelImpl
+import com.revenuecat.purchases.ui.revenuecatui.data.PurchasesImpl
+import com.revenuecat.purchases.ui.revenuecatui.data.PurchasesType
+
+@Composable
+internal fun InternalCustomerCenter(
+    viewModel: CustomerCenterViewModel = getCustomerCenterViewModel(),
+) {
+    CustomerCenterScaffold {
+        when (val state = viewModel.state.collectAsState().value) {
```

Comment:
Suggestion, but not a blocker at all because it's more a matter of taste: 

```diff
-        when (val state = viewModel.state.collectAsState().value) {
+        when (val state by viewModel.state.collectAsState()) {
```